src/tracks/labeltrack/ui/LabelTextHandle.cpp

```diff
@@ -118,21 +105,26 @@ void LabelTextHandle::HandleTextClick(AudacityProject &
             else
                initial = position;
 
-            view.SetTextHighlight( initial, position );
-            mRightDragging = false;
+            view.SetTextSelection(selIndex, initial, position );
          }
          else
+         {
```

Comment:
Very good:  you found out that the right-click case needs handling too.  Maybe just comment `right-click`